test/powershell/Language/Parser/RedirectionOperator.Tests.ps1

```diff
@@ -45,42 +45,36 @@ Describe "Redirection operator now supports encoding changes" -Tags "CI" {
         }
     }
 
-    # $availableEncodings = "unknown","string","unicode","bigendianunicode","utf8","utf7", "utf32","ascii","default","oem"
-    $availableEncodings = (Get-Command Out-File).Parameters["Encoding"].Attributes.ValidValues
-
+    $availableEncodings = 
+        @([System.Text.Encoding]::ASCII,
+          [System.Text.Encoding]::BigEndianUnicode,
+          (New-Object System.Text.UTF32Encoding($true,$true)),
+          [System.Text.Encoding]::Unicode,
+          [System.Text.Encoding]::UTF7,
+          [System.Text.Encoding]::UTF8,
+          [System.Text.Encoding]::UTF32);
+              
     foreach($encoding in $availableEncodings) {
-        $skipTest = $false
-        if ($encoding -eq "default") {
-            # [System.Text.Encoding]::Default is exposed by 'System.Private.CoreLib.dll' at
-            # runtime via reflection. However,it isn't exposed in the reference contract of
-            # 'System.Text.Encoding', and therefore we cannot use 'Encoding.Default' in our
-            # code. So we need to skip this encoding in the test.
-            $skipTest = $true
-        }
 
         # some of the encodings accepted by Out-File aren't real,
         # and Out-File has its own translation, so we'll
         # not do that logic here, but simply ignore those encodings
         # as they eventually are translated to "real" encoding
-        $enc = [System.Text.Encoding]::$encoding
-        if ( $enc )
-        {
-            $msg = "Overriding encoding for Out-File is respected for $encoding"
-            $BOM = $enc.GetPreamble()
-            $TXT = $enc.GetBytes($asciiString)
-            $CR  = $enc.GetBytes($asciiCR)
-            $expectedBytes = .{ $BOM; $TXT; $CR }
-            $PSDefaultParameterValues["Out-File:Encoding"] = "$encoding"
-            $asciiString > TESTDRIVE:/file.txt
-            $observedBytes = Get-Content -AsByteStream TESTDRIVE:/file.txt
-            # THE TEST
-            It $msg -Skip:$skipTest {
-                $observedBytes.Count | Should -Be $expectedBytes.Count
-                for($i = 0;$i -lt $observedBytes.Count; $i++) {
-                    $observedBytes[$i] | Should -Be $expectedBytes[$i]
-                }
+        $encodingName = $encoding.EncodingName
+        $msg = "Overriding encoding for Out-File is respected for $encodingName"
+        $BOM = $encoding.GetPreamble()
+        $TXT = $encoding.GetBytes($asciiString)
+        $CR  = $encoding.GetBytes($asciiCR)
+        $expectedBytes = .{ $BOM; $TXT; $CR }
+        $PSDefaultParameterValues["Out-File:Encoding"] = $encoding
+        $asciiString > TESTDRIVE:/file.txt
+        $observedBytes = Get-Content -AsByteStream TESTDRIVE:/file.txt
+        # THE TEST
+        It $msg {
+            $observedBytes.Count | Should -Be $expectedBytes.Count
+            for($i = 0;$i -lt $observedBytes.Count; $i++) {
+                $observedBytes[$i] | Should -Be $expectedBytes[$i]
```

Comment:
... huh. I guess it's a little out of scope since we're updating an existing test, but the way this is written is _very_ screwy. This should really be refactored to use `-TestCases` instead of a manual loop here, and any setup that has to happen outside the `It` block should go into a `BeforeAll {}` block... Most of it can really happen inside the `It` in any case.

The `for` loop is also unnecessary; Pester is perfectly capable of verifying that all items in two arrays match.

If you'd rather not do a more extensive refactor in this PR, give me a ping when this is merged and I'll come back and refactor the test more thoroughly. 🙂

Reply:
Other changes have been merged.  I'm heading to bed so feel refactor away ;-)